examples/foundational/15-function-calling.py

```diff
@@ -0,0 +1,195 @@
+#
+# Copyright (c) 2024, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+import asyncio
+import aiohttp
+import os
+import json
+import sys
+
+from pipecat.frames.frames import LLMMessagesFrame, Frame
+from pipecat.processors.frame_processor import FrameDirection, FrameProcessor
+from pipecat.pipeline.pipeline import Pipeline
+from pipecat.pipeline.runner import PipelineRunner
+from pipecat.pipeline.task import PipelineTask
+from pipecat.processors.aggregators.llm_response import (
+    LLMAssistantContextAggregator,
+    LLMUserContextAggregator,
+)
+from pipecat.services.openai import OpenAILLMContextFrame, OpenAILLMContext
+from pipecat.processors.logger import FrameLogger
+from pipecat.services.elevenlabs import ElevenLabsTTSService
+from pipecat.services.openai import OpenAILLMService
+from pipecat.transports.services.daily import DailyParams, DailyTransport
+from pipecat.vad.silero import SileroVADAnalyzer
+from openai.types.chat import (
+    ChatCompletionSystemMessageParam,
+    ChatCompletionFunctionMessageParam,
+    ChatCompletionToolParam,
+    ChatCompletionUserMessageParam,
+)
+from pipecat.frames.frames import (
+    LLMFullResponseStartFrame,
+    LLMFullResponseEndFrame,
+    LLMResponseEndFrame,
+    LLMResponseStartFrame,
+    LLMFunctionCallFrame,
+    LLMFunctionStartFrame,
+    TextFrame
+)
+
+from runner import configure
+
+from loguru import logger
+
+from dotenv import load_dotenv
+load_dotenv(override=True)
+
+logger.remove(0)
+logger.add(sys.stderr, level="DEBUG")
+
+
+class FunctionCaller(FrameProcessor):
+    def __init__(self, context):
+        self._context = context
+        super().__init__()
+
+    async def process_frame(self, frame: Frame, direction: FrameDirection):
+        # When we receive function call frames, we need to ask the LLM to run a completion
+        # again so it actually talks to the user
+        if isinstance(frame, LLMFunctionCallFrame):
+            tool_call = ChatCompletionFunctionMessageParam({
+                "role": "assistant",
+                "tool_calls": [
+                    {
+                        "id": frame.tool_call_id,
+                        "function": {
+                            "arguments": frame.arguments,
+                            "name": frame.function_name
+                        },
+                        "type": "function"
+                    }
+                ]
+
+            })
+            self._context.add_message(tool_call)
```

Comment:
I believe this can be added internally when we push `LLMFunctionCallFrame`. There's nothing the user should do in here.